frontend/src/features/manageVehicles/components/list/List.tsx

```diff
@@ -89,6 +89,30 @@ export const List = memo(
     const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
     const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
     const hasNoRowsSelected = Object.keys(rowSelection).length === 0;
+    const queryClient = useQueryClient();
+    const vehicleTypes:VehicleType[] | undefined = queryClient.getQueryData<VehicleType[]>([`${vehicleType}Types`])
+
```

Comment:
Instead of useQueryClient, you can simply use the result from usePowerUnitTypesQuery and useTrailerTypesQuery as mentioned earlier, and get the types through the ".data" field from the result.

Reply:
We usually try to encapsulate the useQueryClient calls inside hooks.ts files rather than using them directly in the code. There are a few instances where it's not encapsulated, I just haven't had the time to refactor them.